packages/firestore/src/remote/remote_store.ts

```diff
@@ -300,39 +295,40 @@ export class RemoteStore implements TargetMetadataProvider {
    */
   private shouldStartWatchStream(): boolean {
     return (
-      this.isNetworkEnabled() &&
+      this.canUseNetwork() &&
       !this.watchStream.isStarted() &&
       !objUtils.isEmpty(this.listenTargets)
     );
   }
 
+  private canUseNetwork(): boolean {
+    // TODO(mikelehen): This could take into account isPrimary when we merge
+    // with multitab.
+    return this.networkEnabled;
+  }
+
   private cleanUpWatchStreamState(): void {
     this.watchChangeAggregator = null;
   }
 
   private async onWatchStreamOpen(): Promise<void> {
-    // TODO(b/35852690): close the stream again (with some timeout?) if no watch
-    // targets are active
     objUtils.forEachNumber(this.listenTargets, (targetId, queryData) => {
       this.sendWatchRequest(queryData);
     });
   }
 
   private async onWatchStreamClose(error?: FirestoreError): Promise<void> {
     assert(
-      this.isNetworkEnabled(),
-      'onWatchStreamClose() should only be called when the network is enabled'
+      error !== undefined || !this.shouldStartWatchStream(),
```

Comment:
I'm not a fan of this assertion here because the condition is hard to reason about. It's better to make the simple assertion that `error !== undefined` inside the if block for `this.shouldStartWatchStream()`.

However, I'm not sure this is a good use of our time. In the caller we're mostly channeling gRPC's response. If it happens to close a stream without supplying an error do we really care? Is it really worth crashing our customers' apps over this?

I'm not arguing against assertions generally: the assertions in the remote store have saved our bacon repeatedly, but the good ones catch our bugs where we failed to anticipate a transition that was actually happening in the wild.

By contrast here it seems like this case of error being null is possible but harmless. If this assertion were to fire in the wild because gRPC was giving us a null error, would we do anything but remove this assertion?

Feel free to push back (especially if I'm not considering all the angles).